### PR TITLE
feat(validator): add tree-shakeable ESM export, fix tests, add a standalone `validator` with zero dependencies

### PR DESCRIPTION
**Pull request by maxkoryukov**
Improve package export and add CJS/ESM versions of a standalone AJV validator for "resume schema" with 0 additional dependencies.

also, the new "export"-design of the package makes "tree shaking" easier with ESM, one only needs to use

```js
import schema from '@jsonresume/schema/schemas/resume.json'  with { type: 'json' }
```
and only the JSON schema will be imported, so the bundler won't include "validate"-functionality in the final bundle (at least from this package). Though, for the "validate.mjs" - the approach is still not ideal. I think another "export" will be required - for "precompiled and standalone" function, and another function that "import ajv" and "import ajv-formats" - so bundler and the programmer can decide how they want to bundle things together.

in order to avoid conflicts - exported as `@jsonresume/schema/all`

the old export and validator - without changes

## Summary by CodeRabbit

* **New Features**
  * Added new package export entrypoints (`./all`, `./validate`, `./schemas/resume.json`, `./schemas/job.json`) for improved modular imports.

* **Chores**
  * Enhanced build pipeline with bundling for both ESM and CommonJS formats.
  * Updated CI/CD workflows to include build verification before tests.
  * Updated development dependencies (ajv, esbuild).
  * Updated .gitignore to exclude build artifacts.

* **Tests**
  * Added comprehensive test coverage for package exports and schema validation across ESM and CommonJS environments.

[Review Change Stack](https://app.coderabbit.ai/change-stack/jsonresume/resume-schema/pull/518?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

----

i've tested like this:

```
import { validate }     from '@jsonresume/schema/all'

// load CV from YAML ...
const cvDataJsonObj = loadCV(12892)

validate(
	cvDataJsonObj,
	function (errs) {
		if (errs) {
			console.error(`CV [${params.id}] was invalid:`, errs)
			return
		}
		console.log(`CV [${params.id}] validated successfully:`)
	},
)
```

### CODE REVIEW

**Review by maxkoryukov — COMMENTED** [review]
self review - DONE

File: tests/resume-awards.test.cjs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+			expect(err).is.null
+			expect(valid).is.true
+			done()
+		})
+	})
+
+	it.only('awards[].title - invalid', function (done) {
+		validate(fixtures.titleInvalid, (err, valid) => {
+console.log(err)
+			expect(err).is.not.null
```
[the remark this answers is not in the document]
Comment (maxkoryukov) [reply tests/resume-awards.test.cjs]: fixed, this file was added by mistake, not it is removed already

**Comment by maxkoryukov** [comment]
@thomasdavis hello, sorry for mentioning directly...

I would be glad to know, if changes in this PR align well with the purpose of the package. If yes - i would improve the PR/add more a few more like this

thanks

**Comment by thomasdavis** [comment]
Thanks for the careful work here — the tree-shakeable exports and the zero-dependency precompiled validator are exactly the right instincts for this package, and the writeup on the export design is genuinely useful.

The reason I'm closing rather than merging: JSON Resume is under active maintenance again (2026), and the project has consolidated into the jsonresume.org monorepo — the schema is moving to packages/schema there, and this repo will be archived with a redirect once triage is done (see jsonresume/jsonresume.org#275). Landing a packaging redesign in a repo that's about to be frozen would just strand it.

I'd love to see this re-proposed against the monorepo once packages/schema exists — the ESM-first export map and standalone validator should inform how that package is published from day one. Sorry for the shuffle, and thanks for the recent burst of contributions.
